src/docfx/lib/markdown/ResolveXref.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Markdig;
+using Markdig.Renderers.Html;
+using Markdig.Syntax.Inlines;
+using Microsoft.DocAsCode.MarkdigEngine.Extensions;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class ResolveXref
+    {
+        public static MarkdownPipelineBuilder UseResolveXref(this MarkdownPipelineBuilder builder, List<DocfxException> errors, Func<string, string> resolveXref)
+        {
+            return builder.Use(document =>
+            {
+                document.Replace(node =>
+                {
+                    if (node is XrefInline xref)
+                    {
+                        var href = resolveXref(xref.Href);
+                        if (string.IsNullOrEmpty(href))
+                        {
+                            var raw = xref.GetAttributes().Properties.First(p => p.Key == "data-raw-source").Value;
+                            var error = raw.StartsWith("@")
```

Comment:
>var error = raw.StartsWith("@") [](start = 28, length = 31)

can judge from "data-throw-if-not-resolved"="true" to log warning

Reply:
didn't know that. I think looking up html attributes for both `raw` and `should-throw `is a hack. I'll add them to markdig ast as part of the reference work.